google-cloud-storage/src/test/scala/akka/stream/alpakka/googlecloud/storage/impl/GoogleCloudStorageClientIntegrationSpec.scala

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.googlecloud.storage.impl
+
+import java.util.UUID
+
+import akka.http.scaladsl.model.ContentTypes
+import akka.stream.alpakka.googlecloud.storage.{StorageSettings, WithMaterializerGlobal}
+import akka.stream.scaladsl.{Sink, Source}
+import akka.util.ByteString
+import com.typesafe.config.{Config, ConfigFactory}
+import org.scalatest._
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.time.{Millis, Seconds, Span}
+
+import scala.collection.immutable.Seq
+import scala.concurrent.Await
+import scala.concurrent.duration._
+import scala.util.Random
+
+/**
+ * USAGE
+ * - Create a google cloud service account
+ * - Make sure it has these roles:
+ *    storage object creator
+ *    storage object viewer
+ *    storage object admin
+ *    storage admin (to run the create/delete bucket test)
+ * - modify test/resources/application.conf to point to the file
+ * - create a bucket for testing
+ * - modify test/resources/application.conf to point to the bucket
+ * - create a rewrite bucket for testing
+ * - modify test/resources/application.conf to point to the bucket
+ */
+class GoogleCloudStorageClientIntegrationSpec
+    extends WordSpec
+    with WithMaterializerGlobal
+    with BeforeAndAfter
+    with Matchers
+    with ScalaFutures {
+
+  private implicit val defaultPatience =
+    PatienceConfig(timeout = Span(60, Seconds), interval = Span(60, Millis))
+
+  var folderName: String = _
+
+  def testFileName(file: String): String = folderName + file
+
+  val config: Config = ConfigFactory.load()
+
+  val bucket = config.getString("alpakka.test.google.storage.bucket")
+  val rewriteBucket = config.getString("alpakka.test.google.storage.rewriteBucket")
+  val projectId = config.getString("alpakka.test.google.storage.projectId")
+  val clientEmail = config.getString("alpakka.test.google.storage.clientEmail")
+  val privateKey = config.getString("alpakka.test.google.storage.privateKey")
```

Comment:
Config support is generally useful, KazeClass has support to create a skeleton for reading settings classes from Config.